include/circt/Dialect/Moore/MooreOps.td

```diff
@@ -454,6 +454,21 @@ def NamedConstantOp : MooreOp<"named_constant", [
   }];
 }
 
+def StringConstantOp : MooreOp<"string", [Pure, ConstantLike]> {
```

Comment:
Minor nit: could we call the op something like `moore.constant_string` or `moore.string_constant` in the assembly? To line up with the C++ name `StringConstantOp` and to have the word "constant" in there :smiley:?